File: src/main/java/com/nkming/powermenu/InstallActivity.java
/*
 * InstallActivity.java
 *
 * Author: Ming Tsang
 * Copyright (c) 2015 Ming Tsang
 * Refer to LICENSE for details
 */

package com.nkming.powermenu;

import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;

public class InstallActivity extends ActionBarActivity
		implements  InstallConfirmFragment.Listener
{
	@Override
	public void onInstallConfirmed()
	{
		InstallFragment f = InstallFragment.create();
		f.show(getSupportFragmentManager(), "install");
	}

	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		if (getIntent().getDataString() != null
				&& getIntent().getDataString().equals(getString(R.string.data_uninstall)))
		{
			onCreateUninstall(savedInstanceState);
		}
		else
		{
			onCreateInstall(savedInstanceState);
		}
	}

	private void onCreateInstall(Bundle savedInstanceState)
	{
		if (savedInstanceState == null)
		{
			InstallConfirmFragment f = InstallConfirmFragment.create();
			f.show(getSupportFragmentManager(), "install_confirm");
		}
	}

	private void onCreateUninstall(Bundle savedInstanceState)
	{
		if (savedInstanceState == null)
		{
			UninstallFragment f = UninstallFragment.create();
			f.show(getSupportFragmentManager(), "uninstall");
		}
	}
}

File: src/main/kotlin/com/nkming/powermenu/InstallActivity.kt
package com.nkming.powermenu

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class InstallActivity : AppCompatActivity(), InstallConfirmFragment.Listener
{
	override fun onInstallConfirmed()
	{
		val f = InstallFragment.create()
		f.show(supportFragmentManager, "install")
	}

	override fun onCreate(savedInstanceState: Bundle?)
	{
		_themeAdapter.onCreate(savedInstanceState)
		super.onCreate(savedInstanceState)
		if (InstallHelper.isSystemApp(this))
		{
			_onCreateUninstall(savedInstanceState)
		}
		else
		{
			_onCreateInstall(savedInstanceState)
		}
	}

	override fun onResume()
	{
		super.onResume()
		_themeAdapter.onResume()
	}

	override fun onPause()
	{
		super.onPause()
		_themeAdapter.onPause()
	}

	override fun onDestroy()
	{
		super.onDestroy()
		_themeAdapter.onDestroy()
	}

	private fun _onCreateInstall(savedInstanceState: Bundle?)
	{
		if (savedInstanceState == null)
		{
			val f = InstallConfirmFragment.create()
			f.show(supportFragmentManager, "install_confirm")
		}
	}

	private fun _onCreateUninstall(savedInstanceState: Bundle?)
	{
		if (savedInstanceState == null)
		{
			val f = UninstallFragment.create()
			f.show(supportFragmentManager, "uninstall")
		}
	}

	private val _themeAdapter by lazy{ActivityThemeAdapter(this,
			R.style.AppTheme_Dark_Dialog, R.style.AppTheme_Light_Dialog)}
}
